src/Evolution/DiscontinuousGalerkin/InboxTags.hpp

```diff
@@ -0,0 +1,94 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <boost/functional/hash.hpp>
+#include <cstddef>
+#include <map>
+#include <optional>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+#include "DataStructures/FixedHashMap.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Structure/MaxNumberOfNeighbors.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Parallel/InboxInserters.hpp"
+#include "Time/TimeStepId.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace evolution::dg::Tags {
+/// The inbox tag for boundary correction communication and DG-subcell ghost
+/// zone cells.
+///
+/// The first element in the tuple is the mesh of the neighboring element's
+/// face (not the mortar mesh!), the second is the volume variables for finite
+/// difference/volume reconstruction, and the third is the fluxes. The fourth
+/// element of the tuple is the range of validity of the boundary terms. This is
+/// need when using local time stepping so that the element knows whether or not
+/// it should remove boundary data and expect new data to be sent from a
+/// neighbor. The TimeStepId also coincides with the next time step of the
+/// neighboring element.
+template <size_t Dim>
+struct BoundaryCorrectionAndGhostCellsInbox {
+  using stored_type =
+      std::tuple<Mesh<Dim - 1>, std::optional<std::vector<double>>,
+                 std::optional<std::vector<double>>, ::TimeStepId>;
```

Comment:
I've tried to document this, but feedback on what I've added will be appreciated/needed since I've been thinking about this code way to much to be a reasonable judge of "This is clear"